### PR TITLE
Add options to refreshBoundingBox to improve performance

### PR DESCRIPTION
**Pull request by bghgary**
Before:
``` js
refreshBoundingInfo(applySkeleton: boolean, applyMorph: boolean)
```

After:
``` js
export interface IMeshDataOptions {
    applySkeleton?: boolean;
    applyMorph?: boolean;
    updatePositionsArray?: boolean;
    cache?: IMeshDataCache;
}

refreshBoundingInfo(options: IMeshDataOptions)
```

The `updatePositionsArray` flag will allow users to disable computing this array which isn't used in all situations.
The `cache` object will cache the allocations, reduce `slice` calls, etc.

On my machine:
[Before](https://playground.babylonjs.com/#B8B8Z2#5): 2.81s
[After](http://localhost:1338/#B8B8Z2#61): 1.67s

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/15234/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15234/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15234/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/15234/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/15234/merge
https://gui.babylonjs.com/?snapshot=refs/pull/15234/merge
https://nme.babylonjs.com/?snapshot=refs/pull/15234/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/15234/merge#BCU1XR#0

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15234/merge/testResults/webgpuplaywright/index.html

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15234/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
Visualization tests for WebGL 1 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15234/merge/testResults/webgl1/index.html

If tests were successful afterwards, this report might not be available anymore.

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15234/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15234/merge/testResults/webgpuplaywright/index.html

**Review by Popov72 — APPROVED** [review]

**Review by sebavan — APPROVED** [review]

**Comment by bghgary** [comment]
I factored out some functions so that native can override. It is much, much faster in native now.

**Comment by bjsplat** [comment]
Visualization tests for WebGL 1 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15234/merge/testResults/webgl1/index.html

If tests were successful afterwards, this report might not be available anymore.

File: packages/dev/core/src/Meshes/abstractMesh.ts [hunk 1]
```diff
@@ -43,6 +43,118 @@ import type { Collider } from "../Collisions/collider";
...
+
+    for (let index = 0, matWeightIdx = 0; index < data.length; index += 3, matWeightIdx += 4) {
+        finalMatrix.reset();
+
+        let inf: number;
+        let weight: number;
+        for (inf = 0; inf < 4; inf++) {
+            weight = matricesWeightsData[matWeightIdx + inf];
+            if (weight > 0) {
+                Matrix.FromFloat32ArrayToRefScaled(skeletonMatrices, Math.floor(matricesIndicesData[matWeightIdx + inf] * 16), weight, tempMatrix);
```
Comment (Popov72) [inline packages/dev/core/src/Meshes/abstractMesh.ts:109]: I'm not sure why the existing code is doing a `floor` here...

**Review by Popov72 — APPROVED** [review]

File: packages/dev/core/src/Meshes/abstractMesh.ts (hunk 1 above)
Comment (bghgary) [reply packages/dev/core/src/Meshes/abstractMesh.ts:109]: I questioned this also, but I'm guessing it can prevent bad data?

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15234/merge/testResults/webgpuplaywright/index.html

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/15234/merge/testResults/webgl2playwright/index.html

**Pull request merged**